docs/concepts/linq.rst

```diff
@@ -0,0 +1,268 @@
+
+.. code-block:: c#
+
+	var dotNetProgrammers = from p in programmers
+	                        where p.IsNewToLINQ
+	                        select new LINQExpert(p);
+	
+Or as a higher-order function via the ``IEnumerable<T>`` interface,
+
+.. code-block:: c#
+
+	var dotNetProgrammers = programmers.Where(p => IsNewToLINQ)
+	                                   .Select(p => new LINQExpert(p));
+									   
+as a way to write expressive, declarative code.
+
+Why is it Useful?
+-----------------
+
+LINQ is an incredibly useful tool for two primary reasons: (1) expressive code, and (2) querying data sources directly from C#/VB via `LINQ Providers <http://stackoverflow.com/a/1568054/1400198>`_  rather than interacting with some obtuse API.
+
+Expressive Code is Valuable
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ask a Ruby programmer why they like Ruby.  Their answer will probably involve something along the lines of "I can express my intent directly in my code".  The same can be said for LINQ.
+
+Imagine you have a list of pets, but want to convert it into a lookup where you can access a pet directly by its ``RFID`` value.
+
+Traditional imperative code:
+
+.. code-block:: c#
+
+	var petLookup = new Dictionary<int, Pet>();
+	
+	foreach (var pet in pets)
+	{
+	    petLookup.Add(pet.RFID, pet);
+	}
+
+Equivalent LINQ expression:
+
+.. code-block:: c#
+
+	var petLookup = pets.ToDictionary(pet => pet.RFIDValue, pet => pet);
+	
+The intention behind the code is not to create a new ``Dictionary<int, Pet>`` and add to it via a loop, it is to convert an existing list into a lookup!  LINQ preserves these semantics whereas the imerative code does not.
+
+Expressive code is valuable because it evens the playing field between intent and code when reasoning as a programmer.  LINQ accomplishes this.
+
+Another bonus is code brevity.  Imagine reducing large portions of a codebase by 1/3 as done above.  Pretty sweet deal, right?
+
+LINQ Providers
+^^^^^^^^^^^^^^
+
+For a significant chunk of software out in the wild, everything revolves around dealing with data from some source (Databases, JSON, XML, etc).  Most APIs which do this can be a pain to work with.  Enter LINQ Providers.
+
+Consider the following: finding all XML elements with a specific attribute value.
+
+.. code-block:: c#
+
+	public static FindAllElementsWithAttribute(XElement documentRoot, string elementName,
+	                                           string attributeName, string value)
+	{
+	    return from el in documentRoot.Elements(elementName)
+	           where (string)el.Element(attributeName) == value
+	           select el;
+	}
+	
+Writing code to manually traverse the XML document to perform this task would be pretty ugly.
```

Comment:
Again, try to avoid "ugly". We have lots of customers writing that code. There is nothing "wrong" with it, but yes, there is a better way.